toolkit/tools/grapher/grapher.go

```diff
@@ -57,7 +72,17 @@ func main() {
 	logger.Log.Info("Running cycle resolution to fix any cycles in the dependency graph")
 	err = depGraph.MakeDAG()
 	if err != nil {
-		logger.Log.Panic(err)
+		if(*resolveCyles) {
```

Comment:
We aim at making sure errors are "pure" - they indicate critical failures, which force us to exit. On rare occasions we also allow ignoring a function error, if we know that it's failure is not a breaking failure for the caller, but that's usually the case with built-in Go libraries dealing with files, which have a well-defined error "interface". One of the strong reasons behind this approach, is that the caller doesn't need to know the internal implementation of the function it's calling, which is required here, since we're making a decision on the error message string. A message, which should have the flexibility to change at any point in time without risking to break any callers.

For the sake of these changes, it looks like a more appropriate approach might be passing `*resolveCycles` (+1 to Bala's comment to re-name this to something, which indicates also **how** we want to resolve these cycles) into `depGraph.MakeDAG`, so that it know what options it has for resolving the cycles.

Reply:
Not really applicable here, but if we do want to check errors a good way to do so is to define a static const error in the package.
As I recall it looks something like this:
```
const MyError = fmt.Errorf("This is my error")`
```

```
innerError := fmt.Errorf("My outer error: %w", myError)
if errors.Is(err, pkg.MyError) {
    ...
}
```

Reply:
removed dependency on error string